tests/jballerina-unit-test/src/test/resources/test-src/statements/vardeclr/uninitialized_module_error_var_decl.bal

```diff
@@ -0,0 +1,27 @@
+//  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+//  WSO2 Inc. licenses this file to you under the Apache License,
+//  Version 2.0 (the "License"); you may not use this file except
+//  in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+type errorDetail record {
+    int riskLevel?;
+};
+
+type riskError error<errorDetail>;
+
+riskError error riskError(errorMsg, cause, riskLevel = riskLevel);
```

Comment:
Even though this is disallowed in the parser itself, shall we add a module `init` function that initialized these variables? It should still fail irrespective of the `init` function.

Reply:
When there is a syntax error we don't reach `DataFlowAnalyzer` right?

Reply:
Even if it gets to the dataflow analyzer it wouldn't be an error, right? 

I meant something like

```ballerina
[int, int] [x, y];

function init() {
    x = 1;
    y = 2;
}
```

Just so that when we look at the test we can say that the declaration itself is wrong.

Reply:
Test cases added